workflow/controller/operator.go

```diff
@@ -1208,19 +1208,20 @@ func getExitCode(pod *apiv1.Pod) *int32 {
 	return nil
 }
 
-func (woc *wfOperationCtx) cleanUpPod(pod *apiv1.Pod) {
+func (woc *wfOperationCtx) cleanUpPod(pod *apiv1.Pod, tmpl wfv1.Template) {
+	var toCleanPod bool
```

Comment:
I'm not a fan of "to" or "do" variables, as they seem to greatly increase the complexity of the code and produce bugs (I'm not sure why, somewhat anecdotal, and I'd need to be a CompSci PHD to produce the analysis). 

Instead, put the logic into its own func. That can then be unit tested.

Reply:
That makes sense. I just moved this to its own function and added unit tests.